✨ Intercepter les codes d'erreurs http, pour se connecter. WIP

manudss opened the pull request.

manudss committed:
✨ Intercepter les codes d'erreurs http, pour se connecter.

patou commented on liste-envies-front/src/app/state/wishes/wishes-list.service.ts:
```diff
@@ -73,10 +81,25 @@ export class WishesListService {
+            case 403:
+              return of(this.router.createUrlTree(["/", "connect"]));
+            default:
+              return of(this.router.createUrlTree(["/"]));
```
Tu n'affiche rien si il y a une erreur ?

patou commented on liste-envies-front/src/app/app-routing.module.ts:
```diff
@@ -44,6 +38,10 @@ const routes: Routes = [
       },
+      {
+        path: "notExist",
+        component: ConnectComponent
```
On n'affiche pas un message personnalisé si il existe pas ? Si on est connecté et que la liste n'existe pas ?

manudss committed:
✨  ajout d'une page si une liste n'hexiste pas.

manudss committed:
✨  pré-remplir la page add list

manudss committed:
✨  ajout not-exist components

manudss committed:
✨  Affichage pop up connection si non connecté + rediction en fonction du type d'erreur.

manudss committed:
refactor code : split and clean code

manudss committed:
:zap: attendre firebase auth soit chargé + ouvrir la wishlist si l'on est déjà connecté

- LE temps que firebase auth soit chargé, les premières requêtes sont lancé (pour chargé la liste des wishlist, ou une wishlist) => mais ces appels ne sont fait avec le token, alors un deuxième appel est fait ensuite.
- l'initialisation faite de firebaseauth permet de prévenir cela avant même que Angular soit chargé. Plus de double requete si l'on est déjà connecté. (pas de pop-up de connection sur une wihlist).
- Maintenant, lors de l'accès d'une liste, tous les cas sont géré. (si non connecté, pop-up de connection (et l'on relance la requete après). Si déjà connecté, on redirige en fonction des cas affichage authorisé de la liste,

manudss committed:
Merge branch 'develop' into feature/intercept-http-errors

patou commented on liste-envies-front/proxy-config.dist.json:
```diff
@@ -1,12 +1,24 @@
+  },
   "/api": {
-    "target": "https://ng5-dot-test-liste-envies.appspot.com",
+    "target": "http://feature-intercept-http-errors-dot-test-liste-envies.appspot.com",
```
```suggestion
    "target": "http://test-liste-envies.appspot.com",
```

patou commented on liste-envies-front/proxy-config.dist.json:
```diff
@@ -1,12 +1,24 @@
     "changeOrigin": true,
     "logLevel": "debug"
   },
   "/user": {
```
Es-ce que ça sert encore le /user, je crois que c'est plus utilisé.

manudss replied:
```suggestion
```

patou commented on liste-envies-front/proxy-config.dist.json:
```diff
@@ -1,12 +1,24 @@
 {
+  "/api2": {
+    "target": "http://liste-envies.desaintsteban.fr",
+    "secure": false,
+    "changeOrigin": true,
+    "logLevel": "debug"
+  },
+  "/api3": {
+    "target": "http://dev-dot-test-liste-envies.appspot.com",
+    "secure": true,
+    "changeOrigin": true,
+    "logLevel": "debug"
+  },
```
```suggestion
```

Supprimer ces lignes qui ne servent plus.

patou commented on liste-envies-front/src/app/app-routing.module.ts:
```diff
@@ -44,6 +38,13 @@ const routes: Routes = [
+      {
+        path: "notExist",
+        loadChildren: () =>
+          import("app/page/not-exists/not-exists.module").then(
```
Très bien de commencer à mettre du lazy loading, mais es-ce que c'est la page la plus grosse ?

manudss replied:
Non c'est pas la plus grosse, mais on peut ne pas en avoir besoins. Donc pas obligé de le chargé avant.

patou commented on liste-envies-front/src/app/state/wishes/wishes-list.service.ts:
```diff
@@ -55,28 +70,139 @@ export class WishesListService {
+    };
+  }
+
+  private throwHttpErrorIfNotAuthorized() {
```
En faite tu simule les code HTTP en fonction des informations sur la liste ?
Je comprend pas pourquoi tu fais ça ? Le serveur doit t'envoyer les statuts en fonction ?

patou commented on liste-envies-front/src/assets/images/undraw_not_found_60pq.svg:
```diff
@@ -0,0 +1 @@
+<svg id="a706cf1c-1654-439b-8fcf-310eb7aa0e00" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" width="1120.59226" height="777.91584" viewBox="0 0 1120.59226 777.91584"><title>not found</title><circle cx="212.59226" cy="103" r="64" fill="#ff6584"/><path d="M563.68016,404.16381c0,151.01141-89.77389,203.73895-200.51559,203.73895S162.649,555.17522,162.649,404.16381,363.16457,61.04208,363.16457 ...
```
Tu as copié le SVG dans le composant.

patou reviewed (commented):
Lorsque l'on clique sur ajouter une liste, les champs ne sont pas initialisés.
https://feature-intercept-http-errors-dot-test-liste-envies.appspot.com/addList?name=apppoline

manudss committed:
Delete undraw_not_found_60pq.svg

manudss commented on liste-envies-front/proxy-config.dist.json:
```diff
@@ -1,12 +1,24 @@
   },
   "/user": {
-    "target": "http://ng5.test-liste-envies.appspot.com",
+    "target": "http://dev-dot-test-liste-envies.appspot.com",
     "secure": false,
     "changeOrigin": true
   }
```
```suggestion
```

manudss committed:
clean dist files

Co-authored-by: Patrice De Saint Steban <patou.de.saint.steban@gmail.com>

manudss committed:
🐛 fix pré-remplissage du nom de la liste sur la page d'ajout.
